File: library/src/main/java/com/getstream/sdk/chat/notifications/NotificationsManager.java
package com.getstream.sdk.chat.notifications;

import android.content.Context;

import com.getstream.sdk.chat.model.Event;
import com.google.firebase.messaging.RemoteMessage;

import org.jetbrains.annotations.NotNull;

/*
 * Created by Anton Bevza on 2019-11-14.
 */
public interface NotificationsManager {

    void setFirebaseToken(@NotNull String firebaseToken, @NotNull Context context);

    void onReceiveFirebaseMessage(@NotNull RemoteMessage remoteMessage, @NotNull Context context);

    void onReceiveWebSocketEvent(@NotNull Event event, @NotNull Context context);

    void handleRemoteMessage(Context context, RemoteMessage remoteMessage);

    void handleEvent(Context context, Event event);

    void setFailMessageListener(NotificationMessageLoadListener failMessageListener);
}

File: library/src/main/java/io/getstream/chat/android/client/notifications/NotificationsManager.kt
package io.getstream.chat.android.client.notifications

import android.content.Context
import com.google.firebase.messaging.RemoteMessage
import io.getstream.chat.android.client.events.ChatEvent

interface NotificationsManager {
    fun setFirebaseToken(
        firebaseToken: String,
        context: Context
    )

    fun onReceiveFirebaseMessage(
        remoteMessage: RemoteMessage,
        context: Context
    )

    fun onReceiveWebSocketEvent(event: ChatEvent, context: Context)

    fun handleRemoteMessage(
        context: Context?,
        remoteMessage: RemoteMessage?
    )

    fun handleEvent(context: Context?, event: ChatEvent?)

    fun setFailMessageListener(failMessageListener: NotificationMessageLoadListener)
}